### PR TITLE
Sgds 244 progressbar

### PR DESCRIPTION
**Pull request by haphizi**
added:
fix elementChildren comp type inference
fix progressbar comp type inference
add unit test
add sb documentation

### CODE REVIEW

File: src/components/ProgressBar/ProgressBar.tsx [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+    className,
+    style,
+    variant,
+    bsPrefix,
+    ...props
+  }: ProgressBarProps,
+  ref: string | ((instance: HTMLDivElement | null) => void) | React.RefObject<HTMLDivElement> | null | undefined
+) {
+  return (
+    <div
```
Comment (clukhei) [inline src/components/ProgressBar/ProgressBar.tsx:159]: use SGDSWrapper ?

File: src/components/ElementChildren.tsx [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+}
+
+/**
+ * Iterates through children that are "valid elements".
+ *
+ * The provided forEachFunc(child, index) will be called for each
+ * leaf child with the index reflecting the position relative to "valid components".
+ */
+function forEach<P = any>(
+  children: boolean | React.ReactChild | React.ReactFragment | React.ReactPortal,
```
Comment (clukhei) [inline src/components/ElementChildren.tsx:29]: without boolean will there be errors? seems like its not a boolean

File: src/components/ElementChildren.tsx [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+/**
+ * Iterates through children that are typically specified as `props.children`,
+ * but only maps over children that are "valid elements".
+ *
+ * The mapFunction provided index will be normalised to the components mapped,
+ * so an invalid component would not increase the index.
+ *
+ */
+function map<P = any>(
+  children: boolean | React.ReactChild | React.ReactFragment | React.ReactPortal,
```
Comment (clukhei) [inline src/components/ElementChildren.tsx:12]: without boolean will there be errors? seems like its not a boolean

File: src/components/ProgressBar/ProgressBar.tsx (hunk 1 above)
Comment (haphizi) [reply src/components/ProgressBar/ProgressBar.tsx:159]: okays

File: src/components/ElementChildren.tsx (hunk 2 above)
Comment (haphizi) [reply src/components/ElementChildren.tsx]: yea no issue

**Review by clukhei — APPROVED** [review]

**Pull request merged**